pulumi/volumeReaper/src/volumeReaper.ts

```diff
@@ -12,6 +12,7 @@ interface StatefulSet extends k8s.V1StatefulSet {
 interface VolumeSnapshot extends Required<k8s.KubernetesObject> {
   metadata: {
     name: string
+    // the actual type is string but the parent object has this messed up
```

Comment:
yeah looking more into this the k8s client has a object serializer for all of the different resources, but with volume snapshots being a crd, there is no defined resource resulting in no deserialization of string -> Date.

we should probably add our own with a comment:
```ts
// no ObjectSerializer exists for VolumeSnapshot CRD, manually deserialize necessary values
const items = body.items.map(item => {
  item.metadata.creationTimestamp = new Date(item.metadata.creationTimestamp)
  return item
})
```

Reply:
Then we can assume whenever we are interacting with `creationTimestamp` thereafter, it is actually a `Date`

Reply:
Added